src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/DigitalGreenCertificateAccess.swift

```diff
@@ -49,16 +50,46 @@ public struct DigitalGreenCertificateAccess: DigitalGreenCertificateAccessProtoc
     }
 
     public func convertToBase45(from base64: Base64, with dataEncryptionKey: Data) -> Result<Base45, CertificateDecodingError> {
+        let cborWebTokenResult = decryptAndComposeToWebToken(from: base64, dataEncryptionKey: dataEncryptionKey)
+        if case let .failure(error) = cborWebTokenResult {
+            return .failure(error)
+        }
+        guard case let .success(cborWebToken) = cborWebTokenResult else {
+            fatalError("Has to be a success at this point, because of previous check.")
+        }
+
+        // Compress with zlib
+        let cborWebTokenData = Data(cborWebToken.encode())
+        let compressedCBORWebToken = cborWebTokenData.compressZLib()
+        guard !compressedCBORWebToken.isEmpty else {
+            return .failure(.HC_ZLIB_COMPRESSION_FAILED)
```

Comment:
Is this error logged later? If not, add a log here? Counts also for any following failure cases.

Reply:
The logger is not available in the scope of the package.
Errors are logged on the caller side.